### PR TITLE
Lighthouse buttons

### PR DESCRIPTION
**Pull request by ns-edisrud**
### Checklist

- [x] All new jobs, commands, executors, parameters have descriptions
- [x] Examples have been added for any significant new features
- [x] README has been updated, if necessary

### Motivation, issues

Over time we will likely be running a handful of different Lighthouse tests and wanted to be able to have slack notifications reflect the results.

### Description
This change allows us to dynamically generate button links to Lighthouse test results from a json.

### CODE REVIEW

File: src/commands/notify.yml [hunk 1]
```diff
@@ -85,6 +86,75 @@ steps:
       name: Slack Notification
       shell: /bin/bash
       command: |
+        # set up potential lighthouse test results
+        curl_arg_1="            -X POST
```
Comment (ns-bdesimone) [inline src/commands/notify.yml]: why all this space?

**Review by ns-bdesimone — COMMENTED** [review]
Few things:
1) I am not so sure i like this whole creating string args and pushing them together. Cant we construct the payload similar to how we were doing it before?
2) Need to update the readme
3) Examples for new features are needed

File: src/commands/notify.yml [hunk 2]
```diff
@@ -67,11 +67,12 @@ parameters:
     type: boolean
     default: true
 
-  dashboard_link:
-    description: The URL to the dashboard
+  lighthouse_button_links:
```
Comment (ns-bdesimone) [inline src/commands/notify.yml]: We are making this generic. Please remove the references to lighthouse

File: src/commands/notify.yml [hunk 3]
```diff
@@ -85,6 +86,64 @@ steps:
       name: Slack Notification
       shell: /bin/bash
       command: |
+        # set up attachments json
+        attachments_json='[{
+        	"fallback": "<< parameters.message >> - $CIRCLE_BUILD_URL",
```
Comment (ns-bdesimone) [inline src/commands/notify.yml]: nit: space

File: src/commands/notify.yml [hunk 4]
```diff
@@ -85,6 +86,64 @@ steps:
...
+        <<# parameters.dashboard_link >>
+        attachments_json=$(echo $attachments_json | jq '.[].actions[.[].actions | length] |= . +
+        {
+        	"type": "button",
+        	"text": "Dashboard",
+        	"url": "<< parameters.dashboard_link >>"
+        }')
+        <</ parameters.dashboard_link >>
+        <<# parameters.lighthouse_button_links >>
+        for link in << parameters.lighthouse_ph_results_link >>; do
```
Comment (ns-bdesimone) [inline src/commands/notify.yml]: more lighthouse refs. These should be generic enough that any report can come in here

**Review by ns-bdesimone — COMMENTED** [review]
🔥 Just remove the lighthouse refs. Then this is good to go!!!!

File: src/commands/notify.yml (hunk 2 above)
Comment (ns-edisrud) [reply src/commands/notify.yml]: done

File: src/commands/notify.yml [hunk 5]
```diff
@@ -67,11 +67,12 @@ parameters:
     type: boolean
     default: true
 
-  dashboard_link:
-    description: The URL to the dashboard
+  test_results_button_links:
```
Comment (ns-bdesimone) [inline src/commands/notify.yml]: Can we just call it button links??? They may not be test results

Comment (ns-edisrud) [reply src/commands/notify.yml]: done

File: src/commands/notify.yml (hunk 3 above)
Comment (ns-edisrud) [reply src/commands/notify.yml]: fixed

File: src/commands/notify.yml [hunk 6]
```diff
@@ -67,11 +67,12 @@ parameters:
     type: boolean
     default: true
 
-  dashboard_link:
-    description: The URL to the dashboard
+  button_links:
+    description: A string (json formatted) with the test names and results links
```
Comment (ns-bdesimone) [inline src/commands/notify.yml]: LOL.... test names -> url names ..... results links -> links

File: README.md [hunk 7]
```diff
@@ -43,7 +43,7 @@ Notify a slack channel with a custom message at any point in a job with this cus
 | `include_project_field` | `boolean` | `true` | Whether or not to include the _Project_ field in the message |
 | `include_job_number_field` | `boolean` | `true` | Whether or not to include the _Job Number_ field in the message |
 | `include_visit_job_action` | `boolean` | `true` | Whether or not to include the _Visit Job_ action in the message |
-| `dashboard_link` | `string` |  | Optional dashboard link property for the [Slack message attachment] |
+| `button_links` | `string` |  | Optional dashboard link property for the [Slack message attachment] (space delimited))|
```
Comment (ns-bdesimone) [inline README.md]: []() is a special md syntax for link. You will have to change that. Also you have an extra ')'

File: src/commands/notify.yml [hunk 8]
```diff
@@ -138,9 +138,9 @@ steps:
         	name="${name%*.html}"
         	attachments_json=$(echo $attachments_json | jq '.[].actions[.[].actions | length] |= . +
         	{
-        		"type": "button",
-        		"text": "'"$name"'",
-        		"url": "'"$link"'"
+              "type": "button",
+              "text": "'"$name"'",
+              "url": "'"$link"'"
```
Comment (ns-bdesimone) [inline src/commands/notify.yml]: One more spacing issue to look at

**Review by ns-bdesimone — APPROVED** [review]
LGTM

**Pull request merged**